tools/bug-report/pkg/cluster/cluster.go

```diff
@@ -298,8 +307,14 @@ func (r *Resources) insertContainer(namespace, deployment, pod, container string
 }
 
 // ContainerRestarts returns the number of container restarts for the given container.
-func (r *Resources) ContainerRestarts(namespace, pod, container string) int {
-	for _, cs := range r.Pod[PodKey(namespace, pod)].Status.ContainerStatuses {
+func (r *Resources) ContainerRestarts(namespace, pod, container string, isCniPod bool) int {
+	var podItem *corev1.Pod
+	if isCniPod {
```

Comment:
do we need this? I thought r.Pod is a superset of r.CniPod

Reply:
I added CniPod as a new field in `Resource` due to there's some pre defined [config](https://github.com/istio/istio/blob/1b2586b8a7739d47e7a0dc7e1e324e66e25e0a3a/tools/bug-report/pkg/filter/filter.go#L71) that will exclude all pods in namespace `kube-system` for `r.Pod` when building the "resource tree" of namespace/pod/container.

Reply:
I see... 

It seems like a bit weird to have CNI code in ContainerRestarts, a bit fragile. Maybe it makes sense to have an AllPods one that isn't filtered... 

Or we can keep this, not a big deal